### ISSUE TITLE
movement shoud be relative to visible part of the molecule

### ISSUE THREAD

**Issue by cryos** [cryos/avogadro#145]
A while ago, Noel proposed that movement when no-atom-is-clicked be relative to the center of the currently visible part of the molecule, rather than relative to the center of the whole molecule.

At that time I thought it was just a wish for cosmetic improvement of the navigation.

But in fact it's really important, it qualifies as a bug report, for the following reason. Open a really large molecule, and navigate to set the camera inside that molecule but not at the center of that molecule. Then try moving forward with 3rd mouse button. Result: the camera actually moves sidewards! (as the center of the molecule is sidewards).

One might think of fixing that by saying that 3rd mouse button makes the camera translate forward instead of toward the molecule center; however that's a bad solution. If we were looking at the molecule from far away, we would want to move toward it, not just move forward in the current camera direction (which could make the molecule fall outside of the view).

In short, we must add code to compute the center of the currently visible part of the molecule (I wrote such pseudocode in a reply to Noel) and use it in the navigation tool.

Reported by: benoitjacob
